README.model_signing.md

```diff
@@ -21,63 +21,79 @@ monitor](https://github.com/sigstore/rekor-monitor) that runs on GitHub Actions.
+against the signed ones.
+
+### Usage
 
-After this, you can clone the repository, create a Python virtual environment
-and install the dependencies needed by the project:
+There are two scripts one can be used to create and sign a bundle and the other to
+verify a bundle. Furthermore, the functionality can be used directly from other
+Python tools. The `sign.py` and `verify.py` scripts can be used as canonical
+how-to examples.
+
+The easiest way to use the scripts directly is from a virtual environment:
 
 ```bash
-git clone git@github.com:sigstore/model-transparency.git
-cd model-transparency/model_signing
-python3 -m venv test_env
-source test_env/bin/activate
-os=Linux # Supported: Linux, Windows, Darwin.
-python3 -m pip install --require-hashes -r "install/requirements_${os}".txt
+$ python3 -m venv .venv
+$ source .venv/bin/activate
+(.venv) $ pip install -r install/requirements.in
 ```
 
-After this point, you can use the project to sign and verify models and
-checkpoints. A help message with all arguments can be obtained by passing `-h`
-argument, either to the main driver or to the two subcommands:
+## Sign
 
 ```bash
-python3 main.py -h
-python3 main.py sign -h
-python3 main.py verify -h
+(.venv) $ python3 sign_model.py --model_path ${MODEL_PATH} --method {sigstore, private-key, pki} {additional parameters depending on method}
 ```
 
-Signing a model requires passing an argument for the path to the model. This can
-be a path to a file or a directory (for large models, or model formats such as
-`SavedModel` which are stored as a directory of related files):
+## Verify
 
 ```bash
-path=path/to/model
-python3 main.py sign --path "${path}"
+(.venv) $ python3 verify_model.py --model_path ${MODEL_PATH} --method {sigstore, private-key, pki} {additional parameters depending on method}
 ```
 
-The sign process will start an OIDC workflow to generate a short lived
-certificate based on an identity provider. This will be relevant when verifying
-the signature, as shown below.
-
-**Note**: The signature is stored as `<file>.sig` for a model serialized as a
-single file, and `<dir>/model.sig` for a model in a folder-based format.
+### Examples
 
-For verification, we need to pass both the path to the model and identity
-related arguments:
+#### Bring Your Own Key
 
 ```bash
-python3 main.py verify --path "${path}" \
-    --identity-provider https://accounts.google.com \
-    --identity myemail@gmail.com
+$ MODEL_PATH='/path/to/your/model'
+$ openssl ecparam -name secp256k1 -genkey -noout -out ec-secp256k1-priv-key.pem
+$ openssl ec -in ec-secp256k1-priv-key.pem -pubout > ec-secp256k1-pub-key.pem
+$ source .venv/bin/activate
+# SIGN
+(.venv) $ python3 sign_model.py --model_path ${MODEL_PATH} --method private-key --private-key ec-secp256k1-priv-key.pem
+...
+#VERIFY
+(.venv) $ python3 verify_model.py --model_path ${MODEL_PATH} --method private-key --public-key ec-secp256k1-pub-key.pem
```

Comment:
Should this have a path to the signature?

Reply:
I see below that you've made that implicit. I'm not sure if we want to go with that, but let's go with this for now.